Library/Homebrew/cask/upgrade.rb

```diff
@@ -40,7 +40,13 @@ def self.upgrade_casks(
 
       quarantine = true if quarantine.nil?
 
-      greedy = true if Homebrew::EnvConfig.upgrade_greedy?
+      if Homebrew::EnvConfig.upgrade_greedy
```

Comment:
Let me know if you prefer a local variable or a empty case statement, e.g. 

```ruby
case Homebrew::EnvConfig.upgrade_greedy
when nil # no-op
when "auto-updates" then greedy_auto_updates = true
when "latest" then greedy_latest = true
else greedy = true
end
```

Reply:
I think we can get rid of this completely by adding some custom methods to `EnvConfig` and add them to the `args`, as you did with your earlier version with `cask_opts_*`.